types/react-transition-group/index.d.ts

```diff
@@ -1,29 +1,15 @@
-// Type definitions for react-transition-group 1.1
+// Type definitions for react-transition-group 2.0
 // Project: https://github.com/reactjs/react-transition-group
 // Definitions by: Karol Janyst <https://github.com/LKay>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-import { HTMLAttributes, ReactElement, ReactType } from "react";
+export { default as CSSTransition, CSSTransitionProps, CSSTransitionClassNames } from "react-transition-group/CSSTransition";
+export { default as Transition, TransitionProps } from "react-transition-group/Transition";
+export { default as TransitionGroup, TransitionGroupProps } from "react-transition-group/TransitionGroup";
```

Comment:
Yes; the problem is the `module.exports = { ... }`. While the internal modules are exported like ES modules, the top level index one isn't and is a regular commonjs export.